enh(chore): Use dayjs instead of moment to save up bundle space and enhance locale handling

bdauria opened the pull request:
## Description

Before:
![Screenshot from 2020-10-29 14-40-36](https://user-images.githubusercontent.com/8367233/97581385-c9020700-19f4-11eb-9151-e336435603ab.png)

After:
![Screenshot from 2020-10-29 14-40-47](https://user-images.githubusercontent.com/8367233/97581388-c99a9d80-19f4-11eb-84a5-b038f9df60c3.png)


**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)

## Target serie

- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

- Make sure all dates and times formatting are correctly handled, including those sent to the API.

Thebarda commented on www/front_src/src/Resources/Actions/api/index.ts:
```diff
@@ -1,6 +1,6 @@
-import formatISO from 'date-fns/formatISO';
 import axios, { AxiosResponse, CancelToken } from 'axios';
-import { map, pick } from 'ramda';
+import { map, pick, startsWith } from 'ramda';
+import dayjs from 'dayjs';
```
dayjs and startsWith are unused here

Thebarda reviewed (commented):
I changed my timezone to Helsinki (+1 hour from Paris)
When selecting a time for a downtime via the Material time picker dialog, It display the time with the wrong timezone (the browser's timezone here)
https://www.loom.com/share/cf091285d439483f91cf8d0195626f16

bdauria commented:
> I changed my timezone to Helsinki (+1 hour from Paris)
> When selecting a time for a downtime via the Material time picker dialog, It display the time with the wrong timezone (the browser's timezone here)
> https://www.loom.com/share/cf091285d439483f91cf8d0195626f16

Fixed

Thebarda reviewed (approved):
Tested and LGTM

kduret commented on www/front_src/src/Provider/index.tsx:
```diff
@@ -1,18 +1,29 @@
 
-import { Provider } from 'react-redux';
-import { pick, pathEq, toPairs, pipe, reduce, mergeAll } from 'ramda';
+import 'dayjs/locale/en';
+import 'dayjs/locale/pt';
+import 'dayjs/locale/fr';
```
missing spanish
